site/content/extend/plugins/developer-workflow.md

```diff
@@ -0,0 +1,66 @@
+- `make reset` - Disables and re-enables the plugin on the Mattermost server
+- `make debug-plugin` - Starts a `delve` process and attaches it to your running plugin
+
+You can run the development build of the plugin by setting the environment variable `MM_DEBUG=1`, or prefixing the variable at the beginning of the `make` command. For example, `MM_DEBUG=1 make deploy` will deploy the development build of the plugin to your server, allowing you to have a more fluid debugging experience. To instead use the production build of the plugin, unset the `MM_DEBUG` environment variable before running the `make` commands.
+
+### Exposing the Mattermost server using `ngrok`
+
+When a public integrates with an external service, webhooks and/or authentication redirects are necessary, which requires your local server to be available on the web. In order for your Mattermost server to be available to process webhook requests, it needs to expose its port to an external address. A common way to do this is to use the command line tool [ngrok](https://ngrok.com). Follow these steps to set up `ngrok` with your server:
+
+- Download the `ngrok` tool from [here](https://ngrok.com/download)
+- Put the executable somewhere within your shell's `PATH`
+- Run `ngrok http 8065` to make your Mattermost server available for webhook requests
+- Set your Mattermost server's [Site URL](http://localhost:8065/admin_console/environment/web_server) to the `https` address given from the `ngrok` command output
+- Monitor webhook requests coming in with ngrok's request inspector. Visit [http://localhost:4040](http://localhost:4040) once you have your tunnel open. You can analyze the contents of the HTTP request from the external service, and the response from your plugin.
+
+If you are using a free `ngrok` account, the URL given by the output of the `ngrok http` command will be different each time you run the command. As a result, you will need to adjust the webhook URL on Mattermost's side and the external service's side (i.e. GitHub) each time you run the command.
+
+With this setup, many integrations require you to be logged into Mattermost using your ngrok URL. After logging into your `ngrok` URL pointed to your Mattermost server, you can in most cases continue using your `localhost` address in your browser for quicker network requests to your server. If you receive an error like `unauthorized` or `enable third-party cookies` when connecting to an external service, make sure you are logged into your `ngrok` URL in the same browser.
+
+### Debugging server-side plugins using `delve`
+
+Using the `make debug-plugin` command will allow you to use a debugger and step through the plugin's server code. A [delve](https://github.com/go-delve/delve) process will be created and attach to your plugin. You can then use an IDE or debug console to connect to the `delve` process. If you are using VSCode, you can use this `launch.json` configuration to connect.
```

Comment:
```suggestion
You can use the `make debug-plugin` command to open a debugger and step through the plugin's server code. A [delve](https://github.com/go-delve/delve) process will be created and attach to your plugin. You can then use an IDE or debug console to connect to the `delve` process. If you're using VSCode, you can use this `launch.json` configuration to connect.
```

Reply:
Note: Not sure if my suggestion changes the meaning of the first sentence.

Reply:
It does change the meaning a bit. The developer has to "open their remote debugger" on their own after running the command.

Reply:
Gotcha. You can disregard my suggestion then. :)